### PR TITLE
Collapse state fix 2

### LINKED ISSUE #219: rework defaults and UI for cell collapse/scrollable/expanded states
I want to rethink how cell collapse state works, particularly WRT presentation mode. Since we allow readers (of view-mode docs) to access the full code behind an analysis just by clicking the 'edit' button, we don't have a strong need to make the JS viewable in-line at all (even as a collapsed pill), so i'm thinking this should be an opt-in feature. that is, by default, state manipulation cells (js, resource, css, etc) should be completely invisible in presentation mode (as should raw cells). The notebook author should be able to make these cells visible if they want to do so for the flow of their narrative, but that should not be the default.

I'm less sure about this part, but I think this also goes for outputs -- they should probably be hidden too, since often in the course of hacking on a notebook you'll want to have intermediate results inline to help you find your way through a problem, but you don't want to have to clean them up for view mode; and in fact, doing could make it harder for people digging into your code in edit mode to follow it. So perhaps the general principle for view mode should be: other than MD and DOM cells (the outputs of which are always shown by default), a NB author has to make an explicit choice to have something shown in View mode. (And conversely, I guess a NB author could choose to hide MD or DOM cells for some reason)

To get the UX to work for these options, we might need to re-introduce a per-cell options menu

see also #144

### LINKED ISSUE #269: Update the evictions notice analysis to apply the modern cell types

### PR DESCRIPTION
**Pull request by bcolloran**
i know i need to get out of the habit of these insanely large PRs, esp now that we have so many contributors, so I apologize for this. Sometimes, tho, it ends up feeling more efficient if I can work safely in a big branch and know that that work will be internally consistent.

In any case, this should fix #269, #224, and #255, which will close out milestone 0.3a entirely (the long awaited JSMD example files are here!)

it also satisfactorily fixes #219, and #144, cleaning up a bunch of stuff around cell rows and cell containers, in the process cleaning up a bit of the corresponding css. there's follow up work to be done in #329 and #330, but it can wait.

once this is merged, i'll be a whisper away from fixing #317 and #315 with a revision of the "what does a web notebook look like" example that I've been working on, and which can be used as the new default intro NB on the GH page. You can see it in the pre-loaded examples. (and once we merge this PR, it will be easy to submit PRs to enhance that and the other examples, b/c they'll be diffable JSMD! woo!)

i'm hoping one of you can can kick the tires on this and we can merge it early tomorrow. All the tests are green, the main thing to do is check out this branch, try the example notebooks, and make sure everything looks how we want.

### CODE REVIEW

**Comment by hamilton** [comment]
Great - will review this once I’m in this morning!

**Comment by hamilton** [comment]
I am testing this branch and get the following error:

screen shot 2018-02-13 at 9 54 07 am

File: src/jsmd-tools.js [hunk 1]
```diff
@@ -111,29 +135,48 @@ function parseJsmd(jsmd) {
...
-  ['javascript', 'js'],
-  ['markdown', 'md'],
-  ['external dependencies', 'resource'],
-  ['raw', 'raw'],
-])
+function stateFromJsmd(jsmdString) {
+  const parsed = parseJsmd(jsmdString)
+  let { cells } = parsed
+  const { parseWarnings } = parsed
+  console.log(parseWarnings)
```
Comment (hamilton) [inline src/jsmd-tools.js]: let's not log parseWarnings unless this array has some contents, and perhaps we use `console.warn` over the contents.

**Comment by hamilton** [comment]
There seems to be a bug in this implementation at the moment that @bcolloran is investigating where new notebook content doesn't show up in presentation mode - @wlach / @mdboom I'd hold off taking a look for now.

File: src/jsmd-tools.js (hunk 1 above)
Comment (bcolloran) [reply src/jsmd-tools.js]: good idea, fixed

**Pull request merged**